portfolio/src/main/webapp/js/custom-script.js

```diff
@@ -116,28 +147,77 @@ function insertSearch() {
 
   const searchInput = document.createElement('input');
   searchInput.className = 'form-control form-inline';
+  searchInput.id = 'searchQuery';
   searchInput.type = 'text';
   searchInput.placeholder = 'Search';
 
   const searchI = document.createElement('i');
   searchI.className = 'fas fa-search text-white ml-3 mr-auto';
 
+  const searchResults = document.createElement('ul');
+  searchResults.id = 'searchResults';
+
   searchDiv.appendChild(searchInput);
   searchDiv.appendChild(searchI);
   searchBar.appendChild(searchDiv);
+  searchBar.append(searchResults);
 
-  document.getElementById('mainNav').appendChild(searchBar);
+  return searchBar;
 }
 
-async function fetchSubmittedLocations() {
-  const response = await fetch('/submitted-locations');
-  const userComments = await response.json();
-  const commentData = [];
+/**
+ * searches each child Node of the page in the docElements and retains the
+   elements that contain the wanted word.
+ */
+function searchPages(docElements, wantedWords) {
+  const numElements = docElements.length;
+  const resultElements = [];
+
+  if (wantedWords.length <= 0) {
+    return resultElements;
+  }
+
+  for (let i = 0; i < numElements; i++) {
+    const frontElement = docElements[i];
+    let frontElementText = frontElement.innerText;
+
+    if (frontElementText) {
+      frontElementText = frontElementText.toLowerCase();
+      if (frontElementText.includes(wantedWords)) {
+        resultElements.push(frontElement);
+      }
+    }
+  }
 
-  userComments.forEach((comment) => {
-    const tempArray = [comment.name, comment.location];
-    commentData.push(tempArray);
-  });
+  return resultElements;
+}
 
-  return commentData;
+/**
+ * populates the search skeleton with the results and sets the functionality to
+   navigate to the result by clicking on it.
+ */
+function showResults(resultElements, wantedWords) {
+  const searchResults = document.getElementById('searchResults');
+  searchResults.innerHTML = '';
+
+  resultElements.forEach((result) => {
+    if (result.getAttribute('aria-hidden') == 'true' ||
+        result.id == 'mainNav') {
+      return;
+    }
+
+    const textElement = document.createElement('li');
+    let resultText = result.innerText.replace(/\s\s+/g, ' ').trim();
+    resultText = resultText.replace(/[\n\r]/g, ' ');
+    const searchPos = resultText.toLowerCase().indexOf(wantedWords);
+    textElement.innerText = '...' +
+        resultText.substring(searchPos - 10, searchPos).replace(/^\s+/g, '') +
+        resultText.substring(searchPos, searchPos + 20).trim() + '...';
```

Comment:
Just checking, does this work properly if the query string is at the start or end of the string?

Reply:
Substrings do not return out of bounds errors so it works in that sense, but it does place an ellipsis regardless of where the string is. That logic could be improved.